test/math/BoolVectorListTest.ooc

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2014 - Simon Mika <simon@mika.se>
+ *
+ * This sofware is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use ooc-math
+use ooc-unit
+import math
+
+BoolVectorListTest: class extends Fixture {
+	init: func {
+		super("BoolVectorList")
+		tolerance := 0.0001f
+		this add("tally and reverse", func {
+			list := BoolVectorList new()
+			list add(true)
+			list add(false)
+			list add(true)
+			list add(true)
+			list add(true)
+			list add(false)
+			expect(list tally(true), is equal to(4))
+			expect(list tally(false), is equal to(2))
+			reversed := list reverse()
+			expect(reversed tally(true), is equal to(list tally(true)))
+			expect(reversed tally(false), is equal to(list tally(false)))
+			expect(!reversed[0])
+			expect(reversed[1])
+			expect(reversed[2])
+			expect(reversed[3])
+			expect(!reversed[4])
+			expect(reversed[5])
+			list free()
+		})
+		
+		this add("dilation", func {
+			list := BoolVectorList new() //1100001110001110
```

Comment:
Would it make sense to add a constructor which takes a cstring of `0` and `1` and creates a vector from it ? It would make this tests so much more readable.